File: 2210. Count Hills and Valleys in an Array/solution.py
class Solution(object):
  def countHillValley(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    newNums = [nums[0]]
    for num in nums[1:]:
      if newNums[-1] != num:
        newNums.append(num)
    
    count = 0
    for index in range(1, len(newNums) - 1):
      before = newNums[index - 1]
      current = newNums[index]
      after = newNums[index + 1]
      if before < current and current > after:
        count += 1
      elif before > current and current < after:
        count += 1
    return count

File: 2210. Count Hills and Valleys in an Array/solution.cpp
class Solution {
public:
  int countHillValley(vector<int>& nums) {
    vector<int> newNums {nums[0]};
    for (int index = 1; index < nums.size(); index++){
      int num = nums[index];
      if (newNums[newNums.size() - 1] != num){
        newNums.push_back(num);
      }
    }

    int count = 0;
    for (int index = 1; index < newNums.size() - 1; index++){
      int before = newNums[index - 1];
      int current = newNums[index];
      int after = newNums[index + 1];
      if (before < current && current > after){
        count += 1;
      }
      else if (before > current && current < after){
        count += 1;
      }
    }
    return count;
  }
};